server/website/website/views.py

```diff
@@ -525,10 +525,16 @@ def handle_result_files(session, files):
         return HttpResponse("Result stored successfully!")
 
     result_id = result.pk
-    response = chain(aggregate_target_results.s(result.pk),
-                     map_workload.s(),
-                     configuration_recommendation.s()).apply_async()
-    taskmeta_ids = [response.parent.parent.id, response.parent.id, response.id]
+    if session.name == 'GPR':
+        response = chain(aggregate_target_results.s(result.pk),
+                         map_workload.s(),
+                         configuration_recommendation.s()).apply_async()
+        taskmeta_ids = [response.parent.parent.id, response.parent.id, response.id]
+    else:
+        response = chain(train_ddpg.s(result.pk),
+                         run_ddpg.s()).apply_async()
+        taskmeta_ids = [response.parent.id, response.id]
+
```

Comment:
Instead of hard-coding taskmeta_ids for every algorithm you can iterate over the tasks:
```
taskmeta_ids = []
current_task = response
while current_task:
    taskmeta_ids.append(current_task.id)
    current_task = current_task.parent
```